File: scripts/download_font.py
#!/usr/bin/env python3
"""
Script to download Reborn Technologies font from Yandex Disk
Run with: python scripts/download_font.py
"""

import os
import sys
import json
import urllib.request
import urllib.parse
from pathlib import Path

def get_download_url(public_url):
    """Get direct download URL from Yandex Disk API"""
    api_url = f"https://cloud-api.yandex.net/v1/disk/public/resources/download?public_key={urllib.parse.quote(public_url)}"
    
    try:
        with urllib.request.urlopen(api_url) as response:
            data = json.loads(response.read().decode())
            return data.get('href')
    except Exception as e:
        print(f"Error fetching download URL: {e}")
        return None

def download_file(url, dest_path):
    """Download file from URL to destination path"""
    try:
        print(f"Downloading from Yandex servers...")
        with urllib.request.urlopen(url) as response:
            data = response.read()
            
        with open(dest_path, 'wb') as f:
            f.write(data)
            
        return len(data)
    except Exception as e:
        print(f"Error downloading file: {e}")
        return None

def main():
    print("=" * 60)
    print("Reborn Technologies Font Downloader")
    print("=" * 60)
    
    # Get script directory
    script_dir = Path(__file__).parent
    project_root = script_dir.parent
    fonts_dir = project_root / "public" / "fonts"
    
    # Create fonts directory if it doesn't exist
    fonts_dir.mkdir(parents=True, exist_ok=True)
    
    dest_path = fonts_dir / "RebornTechnologies.ttf"
    
    print(f"\nStep 1/3: Fetching download URL from Yandex Disk API...")
    public_url = "https://disk.yandex.ru/d/9ri_46n71GxFmg"
    download_url = get_download_url(public_url)
    
    if not download_url:
        print("\n✗ Failed to get download URL")
        print("\nPlease try manual download:")
        print(f"  1. Visit: {public_url}")
        print("  2. Click the Download button")
        print(f"  3. Save to: {dest_path.absolute()}")
        sys.exit(1)
    
    print("✓ Download URL obtained")
    
    print(f"\nStep 2/3: Downloading font file...")
    file_size = download_file(download_url, dest_path)
    
    if not file_size:
        print("\n✗ Failed to download font")
        sys.exit(1)
    
    print(f"✓ Downloaded {file_size:,} bytes")
    
    print(f"\nStep 3/3: Verifying download...")
    
    if dest_path.exists():
        actual_size = dest_path.stat().st_size
        expected_size = 21964
        
        print("\n" + "=" * 60)
        print("✓ Font downloaded successfully!")
        print("=" * 60)
        print(f"File path: {dest_path.absolute()}")
        print(f"File size: {actual_size:,} bytes")
        
        if actual_size == expected_size:
            print(f"Status: ✓ File size matches expected ({expected_size:,} bytes)")
        else:
            print(f"Status: ⚠ File size differs from expected ({expected_size:,} bytes)")
            print(f"        Difference: {actual_size - expected_size:+,} bytes")
        
        print("=" * 60)
        print("\nNext steps:")
        print("  1. Add @font-face rule to your CSS if needed")
        print("  2. Use font-family: 'Reborn Technologies' in your styles")
        print("=" * 60)
    else:
        print("\n✗ File was not created")
        sys.exit(1)

if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        print("\n\nDownload cancelled by user")
        sys.exit(1)
    except Exception as e:
        print(f"\n✗ Unexpected error: {e}")
        import traceback
        traceback.print_exc()
        sys.exit(1)

File: scripts/download-font.js
#!/usr/bin/env node
/**
 * Script to download font from Yandex Disk
 */

const https = require('https');
const http = require('http');
const fs = require('fs');
const path = require('path');

async function getDownloadUrl(publicUrl) {
  const apiUrl = `https://cloud-api.yandex.net/v1/disk/public/resources/download?public_key=${encodeURIComponent(publicUrl)}`;
  
  return new Promise((resolve, reject) => {
    https.get(apiUrl, (res) => {
      let data = '';
      
      res.on('data', (chunk) => {
        data += chunk;
      });
      
      res.on('end', () => {
        try {
          const json = JSON.parse(data);
          resolve(json.href);
        } catch (e) {
          reject(e);
        }
      });
    }).on('error', reject);
  });
}

async function downloadFile(url, destPath) {
  const file = fs.createWriteStream(destPath);
  const urlObj = new URL(url);
  const protocol = urlObj.protocol === 'https:' ? https : http;
  
  return new Promise((resolve, reject) => {
    protocol.get(url, (response) => {
      // Follow redirects
      if (response.statusCode === 301 || response.statusCode === 302) {
        return downloadFile(response.headers.location, destPath)
          .then(resolve)
          .catch(reject);
      }
      
      response.pipe(file);
      
      file.on('finish', () => {
        file.close();
        resolve();
      });
      
      file.on('error', (err) => {
        fs.unlink(destPath, () => reject(err));
      });
    }).on('error', (err) => {
      fs.unlink(destPath, () => reject(err));
    });
  });
}

async function main() {
  try {
    console.log('Fetching download URL from Yandex Disk...');
    const publicUrl = 'https://disk.yandex.ru/d/9ri_46n71GxFmg';
    const downloadUrl = await getDownloadUrl(publicUrl);
    
    console.log('Download URL obtained:', downloadUrl.substring(0, 80) + '...');
    
    const fontsDir = path.join(__dirname, '..', 'public', 'fonts');
    if (!fs.existsSync(fontsDir)) {
      fs.mkdirSync(fontsDir, { recursive: true });
    }
    
    const destPath = path.join(fontsDir, 'RebornTechnologies.ttf');
    console.log('Downloading font to:', destPath);
    
    await downloadFile(downloadUrl, destPath);
    
    const stats = fs.statSync(destPath);
    console.log('\n✓ Font downloaded successfully!');
    console.log('  File:', destPath);
    console.log('  Size:', stats.size, 'bytes');
    
    if (stats.size === 21964) {
      console.log('  ✓ File size matches expected size (21,964 bytes)');
    } else {
      console.log('  ⚠ Warning: File size differs from expected (21,964 bytes)');
    }
  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  }
}

main();
